### ISSUE TITLE
Could you support navigationdestination(item:destination:) ?

### ISSUE THREAD

**Issue by Yoichi-Yamanoi** [johnpatrickmorgan/NavigationBackport#49]
Hi.
I want to use your library but my app support iPhone and iPad.
I depend on [navigationDestination(item:destination:)](https://developer.apple.com/documentation/swiftui/view/navigationdestination(item:destination:)) for screen transitions.

Could you support it ?

```swift
@State private var colorShown: Color?


NavigationSplitView {
    List {
        Button("Mint") { colorShown = .mint }
        Button("Pink") { colorShown = .pink }
        Button("Teal") { colorShown = .teal }
    }
    .nbNavigationDestination(item: $colorShown) { color in
        ColorDetail(color: color)
    }
} detail: {
    Text("Select a color")
}

```

**Comment by Yoichi-Yamanoi** [comment]
I can achieve what I want to do by using the following approach.
```swift
extension View {
    func nbNavigationDestination<D: Hashable, C: View>(item: Binding<D?>, @ViewBuilder destination: @escaping (D) -> C) -> some View {
        nbNavigationDestination(
            isPresented: Binding(
                get: { item.wrappedValue != nil },
                set: { isActive, transaction in
                    if !isActive {
                        item.transaction(transaction).wrappedValue = nil
                    }
                }
            ),
            destination: { item.wrappedValue.map(destination) }
        )
    }
}
```

**Comment by simba909** [comment]
@Yoichi-Yamanoi san that seems like a great approach! I think you should open a PR with that addition 👍🏻

**Comment by Yoichi-Yamanoi** [comment]
Thanks @simba909 !  I created [PR](https://github.com/johnpatrickmorgan/NavigationBackport/pull/52) so I would be happy if you could review it.

**Comment by simba909** [comment]
Your PR looks great! I can't approve it though as I'm not a collaborator of this repository. I'll leave approval / merging to @johnpatrickmorgan

**Comment by johnpatrickmorgan** [comment]
Thanks @Yoichi-Yamanoi! The PR is now merged and this was shipped in v0.9.3.
